pkg/nodehandler/controller.go

```diff
@@ -229,9 +231,47 @@ func addNodeRemovalLabel(csibmnode *nodecrd.Node) {
 	if csibmnode.Labels == nil {
 		csibmnode.Labels = map[string]string{}
 	}
-	csibmnode.Labels[nodeRemovalTaintKey] = nodeRemovalTaintValue
+	csibmnode.Labels[rTaint.Key] = rTaint.Value
 }
 
 func deleteNodeRemovalLabel(csibmnode *nodecrd.Node) {
-	delete(csibmnode.Labels, nodeRemovalTaintKey)
+	delete(csibmnode.Labels, rTaint.Key)
+}
+
+func (c *Controller) handleNodeMaintenance(ctx context.Context, nodes []corev1.Node) error {
+	for i, node := range nodes {
+		if hasTaint(&nodes[i], mTaint) {
+			if err := c.deleteCSIPods(ctx, node.Name); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Controller) deleteCSIPods(ctx context.Context, nodeName string) error {
+	c.log.Info(fmt.Sprintf("Starting to delete CSI pods for on node %s", nodeName))
+
+	fieldSelector := fields.SelectorFromSet(map[string]string{"spec.nodeName": nodeName})
+	labelSelector := labels.SelectorFromSet(common.ConstructLabelAppMap())
```

Comment:
Should we delete daemonset pods like csi-baremetal-node or csi-baremetal-se? They have this label too

Reply:
K8s restart them anyway and in my opinion we can have races during volumes deleting in the same time by another operators

Reply:
Effect NoSchedule forbids k8s creating pods on tainted nodes. And expected killing all pods on tainted nodes.
Is expected that another operators will be delete volumes for entering maintenance mode?

Reply:
CSI controls volumes from another pods. And as I understand behavior, they must be removed too. If we delete csi-baremetal-node, it won't be able to unmount and delete volumes, is it?

Reply:
Lets don't delete daemonset pods

Reply:
Done